lib/temporal_tables/temporal_class.rb

```diff
@@ -74,8 +74,10 @@ def orig_class
 
       delegate :descends_from_active_record?, to: :superclass
 
+      # An object at a given time should fall within the range, excluding the effective end date.
+      # However, when using '9999-12-31', this is effectively infinity and should not be excluded.
       def build_temporal_constraint(at_value)
-        arel_table[:eff_to].gteq(at_value).and(
+        (arel_table[:eff_to].gt(at_value).or(arel_table[:eff_to].eq('9999-12-31'))).and(
```

Comment:
This is a bit hacky but works. I wonder if the better way of doing this is that in `association_extensions.rb`, instead of using `@owner.eff_to`, it should use the current time. Depending on how people use the library though (like manually setting effective ranges in the future), that could be a more significant breaking change.